### DISCUSSION TITLE
Feature request: bulk delete old branches

### DISCUSSION THREAD

**Discussion by nathanl** [community/community#68425] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

It is common for repos to accumulate dead branches. **These are clutter for developers and also force GitHub to retain the commits on those branches in storage**. `git branch -a --merged` will not identify them if they were squash merged. In that case, deleting them requires a human judgment call. Right now, this requires tedious `git push origin :dead_branch_name` commands.

I request a "branch cleanup" tool that lists all branches by the age of their last commit. For each branch, info about the last commit (message, committer, date) should be shown. Users could check a box next to each branch to be deleted and bulk delete them.

As evidence that this feature would be popular, the StackOverflow question [How do I remove all stale branches from Github?](https://stackoverflow.com/questions/39065544/how-do-i-remove-all-stale-branches-from-github) has been viewed 22k times.

I believe this feature would also free up storage space for GitHub as many old, irrelevant commits could then be removed with `git gc`.

**Comment by nathanl** [comment, marked as answer]
I just realized that `/branches` on any repo provides a reasonable way to do this.

**Reply by mahesh-attarde** [reply]
What does that mean "/branches on any repo"? on URL? 
On listing, it is still one by one delete. Did you find way to bulk delete?

**Reply by nathanl** [reply]
Yes, `https://github.com/orgname/reponame/branches`. It is still 1 click per delete though. I don't know a way to do it faster, but they have tabs like "Yours" and "Stale" that make identifying candidates for deletion easier.

**Reply by mahesh-attarde** [reply]
I see. Thanks.

**Reply by taylorjayoung** [reply]
@nathanl @mahesh-attarde Do you think this is worth building? I built a tool for bulk deleting (reposweeper.com) and could easily add this.

**Reply by taylorjayoung** [reply]
This is implemented on reposweeper.com now. Just basic select branches in bulk and click delete. But lmk if you have any feedback or suggestions for other bulk branch management features. Cheers @nathanl @mahesh-attarde  

<img width="1236" height="725" alt="ManageBranches" src="https://github.com/user-attachments/assets/cfec773d-4550-45ad-9424-8ab24981565f" />


<img width="921" height="782" alt="Delete Instructions" src="https://github.com/user-attachments/assets/365a8b3c-b905-41a5-a02c-362b99c7c4b0" />
